File: Solutions/[08]_5.py
n, m = map(int, input().split())
array = []

for _ in range(n):
    array.append(input())

row = [0] * n
column = [0] * m

for i in range(n):
    for j in range(m):
        if array[i][j] == 'X':
            row[i] = 1
            column[j] = 1

row_count = 0
for i in range(n):
    if row[i] == 0:
        row_count += 1

column_count = 0
for j in range(m):
    if column[j] == 0:
        column_count += 1

print(max(row_count, column_count))

File: Solutions/[08]_5.java
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		String input[] = sc.nextLine().split(" ");
		int n = Integer.parseInt(input[0]);
		int m = Integer.parseInt(input[1]);
		
		char[][] data = new char[50][50];
		int[] row = new int[50];
		int[] column = new int[50];

		for (int i = 0; i < n; i++) {
			data[i] = sc.nextLine().toCharArray();
			for (int j = 0; j < m; j++) {
				if (data[i][j] == 'X') {
					row[i] = 1;
					column[j] = 1;
				}
			}
		}

		int row_count = 0;
		for (int i = 0; i < n; i++) {
			if (row[i] == 0) row_count++;
		}

		int column_count = 0;
		for (int j = 0; j < m; j++) {
			if (column[j] == 0) column_count++;
		}

		System.out.println(Math.max(row_count, column_count));
	}

}
